pkg/storage/swift/utils.go

```diff
@@ -0,0 +1,71 @@
+package swift
+
+import (
+	"fmt"
+
+	"github.com/gophercloud/utils/openstack/clientconfig"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	coreset "k8s.io/client-go/kubernetes/typed/core/v1"
+
+	imageregistryv1 "github.com/openshift/cluster-image-registry-operator/pkg/apis/imageregistry/v1"
+	regopclient "github.com/openshift/cluster-image-registry-operator/pkg/client"
+
+	yaml "gopkg.in/yaml.v2"
+)
+
+// getSwiftCredentialsFromSecret reads credentials from the operator's native secret
+func getSwiftCredentialsFromSecret(listers *regopclient.Listers) (credentials, error) {
+	creds := credentials{}
+
+	// Look for a user defined secret to get the Swift credentials
+	sec, err := listers.Secrets.Get(imageregistryv1.ImageRegistryPrivateConfigurationUser)
```

Comment:
You need to save data from the `ImageRegistryPrivateConfigurationUser` to `ImageRegistryPrivateConfiguration` as well.

Reply:
I don't quite get it... Do you want me to copy data from ImageRegistryPrivateConfigurationUser secret to ImageRegistryPrivateConfiguration secret?

Reply:
Okay, done!